### PR TITLE
[WX-1758] Implement deleting method snapshots from methods repository

### PR DESCRIPTION
**Pull request by sam-schu**
### What
- Added a kebab (vertical three-dot) menu to the method details page in the Methods Repository.
- Added a button in the menu to open a "Delete snapshot" modal. The button is disabled and shows a tooltip if the user does not have permission to delete the method snapshot being viewed (i.e., if they are not an owner of the snapshot).
  - In Firecloud UI, the button does not appear at all in this case, but the behavior implemented here is more consistent with other parts of Terra UI (such as the kebab menu for workspace actions).
- Implemented the modal to delete a snapshot from the Methods Repo (with error handling).
  - When the user confirms that they want to delete the snapshot, they are automatically brought back to the methods list page, and the navigation history and stores are updated to achieve the following behavior:
    - If the user clicks on the same method in the methods list or presses the back button in their browser to return to the method that they just deleted a snapshot of, they will be automatically redirected to the most recent/highest-numbered snapshot of that method that still exists, rather than the one that they just deleted, and the deleted snapshot will not appear in the snapshot dropdown.
    - If the user manually navigates to the URL of the old snapshot, outdated cached information about the snapshot will not be shown. ([WX-1857](https://broadworkbench.atlassian.net/browse/WX-1857) will improve the error handling in this case.)

### Why
- To continue the Firecloud UI to Terra UI migration by allowing users to delete (formerly, "redact") a method snapshot from the Methods Repository within Terra UI. (This feature may be used, for example, if a user discovers that the WDL for a method snapshot they published is incorrect.)

### Testing strategy
- [x] Manual testing
- [x] Unit tests

### Visual Aids
Method details page with kebab menu:
method details page image

Delete snapshot button when the user does not have permission:
method details page for non-owner image

Error message due to failed network request:
method details page with delete snapshot error image

Delete snapshot modal:
Terra UI delete snapshot modal image

Firecloud UI delete snapshot modal for reference:
Firecloud UI delete snapshot modal image

[WX-1857]: https://broadworkbench.atlassian.net/browse/WX-1857?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/pages/workflows/workflow/WorkflowWrapper.ts [hunk 1]
```diff
@@ -121,6 +134,31 @@ export const WorkflowsContainer = (props: WorkflowContainerProps) => {
...
+  const deleteSnapshot = async () => {
+    await Ajax(signal).Methods.method(namespace, name, selectedSnapshot).delete();
+
+    // Replace the current history entry linking to the method details page of a
+    // specific snapshot, like /#workflows/sschu/echo-strings-test/29, with an
+    // entry with the corresponding link without the snapshot ID, like
+    // /#workflows/sschu/echo-strings-test
+    // This way, if the user presses the back button after deleting a
+    // method snapshot, they will be automatically redirected to the most recent
+    // snapshot that still exists of the same method
```
Comment (kpierre13) [inline src/pages/workflows/workflow/WorkflowWrapper.ts:146]: Great comments!

File: src/workflows/SnapshotActionMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ * Currently supported actions: delete
+ */
+const SnapshotActionMenu = (props: SnapshotActionMenuProps): ReactNode => {
+  const { isSnapshotOwner, onDelete } = props;
+
+  const notSnapshotOwnerTooltip = 'You must be an owner of this snapshot';
+
+  const menuContent = (
+    <MenuButton
+      disabled={!isSnapshotOwner}
```
Comment (kpierre13) [inline src/workflows/SnapshotActionMenu.tsx:30]: Out of curiosity, is there a reason why you decided to disable the delete option as opposed to hiding it all together?

**Review by kpierre13 — APPROVED** [review]
Just one question, but LGTM

File: src/workflows/SnapshotActionMenu.tsx (hunk 2 above)
Comment (sam-schu) [reply src/workflows/SnapshotActionMenu.tsx:30]: I did this for consistency with other kebab menus in Terra UI, such as for workspace actions:

disabled delete workspace action image

File: src/libs/ajax/methods/Methods.ts [hunk 3]
```diff
@@ -40,6 +40,11 @@ export const Methods = (signal?: AbortSignal) => ({
         return res.json();
       },
 
+      delete: async () => {
+        const res = await fetchOrchestration(`api/${root}`, _.merge(authOpts(), { signal, method: 'DELETE' }));
```
Comment (salonishah11) [inline src/libs/ajax/methods/Methods.ts:44]: It seems the other methods here directly call Agora. Is there a reason to go through Orchestration for this API?

Comment (kpierre13) [reply src/libs/ajax/methods/Methods.ts:44]: There is! @sam-schu had asked a question in Slack about whether to use Agora or Orch endpoints, and Adam mentioned that Orch is in fact the intended API as opposed to Agora.

Comment (sam-schu) [reply src/libs/ajax/methods/Methods.ts:44]: Firecloud UI makes this API call to Orchestration, and in a [Slack discussion](https://broadinstitute.slack.com/archives/C06NRD09KBR/p1726075469016359) with @aednichols, I was advised that calling Orchestration is preferred to calling services like Agora directly (and there is a chance that Orchestration will modify the request before making the call to Agora).

Comment (salonishah11) [reply src/libs/ajax/methods/Methods.ts:44]: Thank you for the clarification.

**Review by salonishah11 — APPROVED** [review]

**Pull request merged**